src/main/java/games/strategy/engine/lobby/server/db/BadWordController.java

```diff
@@ -6,27 +6,20 @@
 import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.logging.Logger;
 
 /**
  * Utilitiy class to create/read/delete bad words (there is no update).
  */
 public class BadWordController {
-  private static final Logger logger = Logger.getLogger(BadWordController.class.getName());
-
   public void addBadWord(final String word) {
     try (final Connection con = Database.getPostgresConnection();
-        final PreparedStatement ps = con.prepareStatement("insert into bad_words (word) values (?)")) {
+        // If the word already is present we don't need to add it twice.
+        final PreparedStatement ps =
+            con.prepareStatement("insert into bad_words (word) values (?) on conflict do nothing")) {
```

Comment:
A couple of integration tests (one for the happy path and one for the duplicate primary key) for each of these _Table Data Gateway_ classes (e.g. `BadWordController`, `BannedMacController`, etc.) would probably be a good idea.

Reply:
@ssoloff Would work in this case, but the other classes have some logic in them wich prevents the SQL from being executed if the entry is already there,

Reply:
Yeah, we need to consider modifying our DB integration tests to run within transactions that are rolled back at the end of each test so subsequent tests aren't affected by previous state.  They are very brittle at the moment for devs.  Every once in a while I have to blow away my DB to work around this kind of problem.  It's not too hard with a Docker container, but it still smells. :smile:

Reply:
I added integration test for the other classes, once I figured I could remove the logic.
Agreed, it's really not optimal, but recreating the db is not that hard even without using docker.